src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -24,11 +24,11 @@ public class IPAddress
 
         internal const uint LoopbackMaskHostOrder = 0xFF000000;
 
-        public static readonly IPAddress IPv6Any = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
-        public static readonly IPAddress IPv6Loopback = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }, 0);
-        public static readonly IPAddress IPv6None = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
+        public static readonly IPAddress IPv6Any = new ReadOnlyIPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
+        public static readonly IPAddress IPv6Loopback = new ReadOnlyIPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }, 0);
+        public static readonly IPAddress IPv6None = new ReadOnlyIPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
 
-        private static readonly IPAddress s_loopbackMappedToIPv6 = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1 }, 0);
+        private static readonly IPAddress s_loopbackMappedToIPv6 = new ReadOnlyIPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1 }, 0);
```

Comment:
Why do these types need to change?  Couldn't this just be done by adding a `(ReadOnlySpan<byte>)` cast before each `new byte[]`?

Reply:
I did it for symmetry with the IPv4 fields (yes, I know there is a functional reason for these to be of a different type), and for clarity to avoid these casts. But I can bring them instead if there's a problem.

Reply:
The derived type isn't needed for IPv6, and there's a small risk of breaking code by changing the type (we got a few complaints when it was done for the IPv4 instances). Since we don't _need_ to here, we shouldn't.

Reply:
Oh, I see, I changed it.